wgpu-core/src/device/global.rs

```diff
@@ -2336,15 +2336,47 @@ impl Global {
     pub fn buffer_map_async<A: HalApi>(
         &self,
         buffer_id: id::BufferId,
-        range: Range<BufferAddress>,
+        offset: BufferAddress,
+        size: Option<BufferAddress>,
         op: BufferMapOperation,
     ) -> BufferAccessResult {
-        api_log!("Buffer::map_async {buffer_id:?} range {range:?} op: {op:?}");
+        api_log!("Buffer::map_async {buffer_id:?} offset {offset:?} size {size:?} op: {op:?}");
+
+        let hub = A::hub(self);
```

Comment:
Let's put the code below in `buffer_map_async_inner`, to put the validation in once place and so that we don't have to get the buffer from the registry twice.

Reply:
I think we need to change the signature of `buffer_map_async_inner` as well then to pass in the buffer. Otherwise the we cannot get the actual `range_size`

Reply:
Let's make `buffer_map_async_inner` take an offset and an optional size.

Reply:
Done! Sorry for the long thread for such a small change.

I am still picking up Rust, so appreciate any comments!

Reply:
No worries, I was the one who came with conflicting requirements in the first place.